sdks/python/apache_beam/runners/direct/direct_runner.py

```diff
@@ -79,17 +87,24 @@ def apply_CombinePerKey(self, transform, pcoll):
     except NotImplementedError:
       return transform.expand(pcoll)
 
+  def apply__GroupByKeyOnly(self, transform, pcoll):
+    if (transform.__class__ == _GroupByKeyOnly and
```

Comment:
Yes--because we subclass _GroupByKeyOnly in _StreamingGroupByKeyOnly, this would also be called for expanding _StreamingGroupByKeyOnly.  This check lets us avoid recursive calls here.